### PR TITLE
providing fix for https://github.com/krakenjs/makara/issues/76

### PR DESCRIPTION
**Pull request by grawk**
allows `@provide` to work with makara2 `dust-message-helper` in "paired" mode.

See: https://github.com/krakenjs/makara/blob/v2.x/ADVANCED.md
Resolves: https://github.com/krakenjs/makara/issues/76

### CODE REVIEW

File: src/helpers/data/provide/provide.js [hunk 1]
```diff
@@ -7,36 +7,66 @@
...
+
+      // Render all variable bodies, and store the resulting values
+      var values = {};
+      for (var key in bodies) {
+        if (key !== 'block') {
+          var data = getData(bodies[key]);
+          try {
+            values[key] = JSON.parse(data);
+          } catch (error) {
+            values[key] = data;
```
Comment (shaunwarman) [inline src/helpers/data/provide/provide.js:35]: I imagine rare, but will something handle the JSON later?

Comment (grawk) [inline src/helpers/data/provide/provide.js:35]: Are you asking about the case where `JSON.parse` throws an error?

Comment (shaunwarman) [inline src/helpers/data/provide/provide.js:35]: Ya just wondering if it's safe to add to `values[key]` regardless.

Comment (grawk) [inline src/helpers/data/provide/provide.js:35]: It's essentially what's already done in the existing code:
https://github.com/krakenjs/festucam/blob/master/src/helpers/data/provide/provide.js#L30

There's a unit test to describe the behavior in this scenario:
https://github.com/krakenjs/festucam/blob/master/test/provide-test.js#L71

Comment (shaunwarman) [inline src/helpers/data/provide/provide.js:35]: Ah perfect, ya https://github.com/krakenjs/festucam/blob/master/src/helpers/data/provide/provide.js#L30

**Comment by shaunwarman** [comment]
👍

**Pull request merged**